core/src/main/java/com/scalar/db/api/TwoPhaseCommitTransactionManager.java

```diff
@@ -108,19 +120,25 @@ default TwoPhaseCommitTransaction start(String txId) throws TransactionException
    *
    * @param txId the transaction ID
    * @return {@link TwoPhaseCommitTransaction}
+   * @throws RetriableTransactionException if something retriable situation happens. You can retry
+   *     the same transaction from the beginning in this case
    * @throws TransactionException if participating the transaction fails
    */
-  TwoPhaseCommitTransaction join(String txId) throws TransactionException;
+  TwoPhaseCommitTransaction join(String txId)
+      throws RetriableTransactionException, TransactionException;
 
   /**
    * Resumes an ongoing transaction associated with the specified transaction ID.
    *
    * @param txId the transaction ID
    * @return {@link TwoPhaseCommitTransaction}
    * @throws TransactionNotFoundException if the transaction associated with the specified
-   *     transaction ID is not found
+   *     transaction ID is not found. You can retry the same transaction from the beginning in this
+   *     case
+   * @throws TransactionException if resuming the transaction fails
    */
-  TwoPhaseCommitTransaction resume(String txId) throws TransactionNotFoundException;
+  TwoPhaseCommitTransaction resume(String txId)
+      throws TransactionNotFoundException, TransactionException;
```

Comment:
Made `resume()` throw `TransactionNotFoundException` and `TransactionException`. Actually, it's incompatible change, but this `resume()` is not used widely, so I think it's okay.